File: src/main/java/ru/citeck/ecos/uiserv/domain/journal/repo/JournalEntity.java
package ru.citeck.ecos.uiserv.domain.journal.repo;

import lombok.Data;
import ru.citeck.ecos.uiserv.domain.common.repo.AbstractAuditingEntity;

import javax.persistence.*;

@Entity
@Table(name = "journal")
@Data
public class JournalEntity extends AbstractAuditingEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "journal_seq_gen")
    @SequenceGenerator(name = "journal_seq_gen")
    private Long id;

    @Column(unique = true)
    private String extId;

    private String label;

    private String sourceId;

    private String groupBy;

    private String sortBy;

    private String metaRecord;

    private String typeRef;

    private String predicate;

    private Boolean editable;

    private String attributes;

    private String columns;

    private String actions;

    private String groupActions;
}

File: src/main/java/ru/citeck/ecos/uiserv/domain/journal/repo/JournalEntity.kt
package ru.citeck.ecos.uiserv.domain.journal.repo

import jakarta.persistence.*
import ru.citeck.ecos.uiserv.app.common.repo.AbstractAuditingEntity

@Entity
@Table(name = "journal")
class JournalEntity : AbstractAuditingEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "hibernate_sequence")
    @SequenceGenerator(name = "hibernate_sequence")
    val id: Long? = null

    @Column(unique = true)
    lateinit var extId: String
    lateinit var columns: String

    var sourceId: String? = null
    var metaRecord: String? = null

    @Column(name = "label")
    var name: String? = null
    var groupBy: String? = null
    var sortBy: String? = null
    var typeRef: String? = null
    var workspace: String = ""
    var predicate: String? = null
    var defaultFilters: String? = null
    var queryData: String? = null
    var searchConfig: String? = null
    var editable: Boolean? = null
    var attributes: String? = null
    var hideImportDataActions: Boolean? = null
    var actionsFromType: Boolean? = null
    var actions: String? = null
    var actionsDef: String? = null
    var computed: String? = null

    var system: Boolean? = null
}
